pyartcd/pyartcd/pipelines/promote.py

```diff
@@ -879,45 +858,47 @@ async def get_image_info(pullspec: str, raise_if_not_found: bool = False):
                     raise IOError(f"Image {pullspec} is not found.")
                 return None
             raise ChildProcessError(f"Error running {cmd}: exit_code={rc}, stdout={stdout}, stderr={stderr}")
+
+        # Info provided by oc need to be converted back into Skopeo-looking format
         info = json.loads(stdout)
-        if not isinstance(info, dict):
+        if not isinstance(info, list):
             raise ValueError(f"Invalid image info: {info}")
-        return info
+
+        media_types = set([manifest['mediaType'] for manifest in info])
+        if len(media_types) > 1:
+            raise ValueError(f'Inconsistent media types across manifests: {media_types}')
+
+        manifests = {
+            'mediaType': media_types.pop(),
+            'manifests': [
+                {
+                    'digest': manifest['digest'],
+                    'platform': {
+                        'architecture': manifest['config']['architecture'],
+                        'os': manifest['config']['os']
+                    }
+                } for manifest in info
+            ]
+        }
+
+        return manifests
 
     @staticmethod
     async def get_image_digest(pullspec: str, raise_if_not_found: bool = False):
         # Get image digest
-        # We use skopeo instead of `oc image info` because oc doesn't support json/yaml output for a manifest list.
-        if "://" not in pullspec:
-            pullspec = f"docker://{pullspec}"
-        # skopeo on buildvm is too old. Use a containerized version instead.
-        cmd = [
-            "podman",
-            "run",
-            "--rm",
-            "--privileged",
-            "quay.io/containers/skopeo:v1.11"
-        ]
-        if os.environ.get("PYARTCD_USE_NATIVE_SKOPEO") == "1":
-            cmd = ["skopeo"]
-        cmd.extend([
-            "--override-os=linux",  # needed to run this command on macOS
-            "inspect",
-            "--no-tags",
-            "--format={{.Digest}}",
-            "--",
-            pullspec,
-        ])
+        cmd = f'oc image info {pullspec} --filter-by-os linux/amd64 -o json'
```

Comment:
I don't see a reference to amd64 in the skopeo invocation.. wondering how it was working before. 
If this func is used to check only multi amd64 then we should rename it to indicate that

Reply:
Are we always passing a manifest list to this func? If we pass in `4.12.10-x86_64` skopeo would work but oc would fail here I think

Reply:
This function is called [here](https://github.com/openshift-eng/aos-cd-jobs/blob/master/pyartcd/pyartcd/pipelines/promote.py#L659) and [here](https://github.com/openshift-eng/aos-cd-jobs/blob/master/pyartcd/pyartcd/pipelines/promote.py#L741).

In both cases, we pass in `f"quay.io/openshift-release-dev/ocp-release:{release_name}-multi"`.

Reply:
> Ah so listDigest is the manifest list digest.. it is the same across all arches

yep!

Reply:
Done in [34c586e](https://github.com/openshift-eng/aos-cd-jobs/pull/3603/commits/34c586e2d01af1ddbc6ca9ebd61b5cdaeecf49f2)